### PR TITLE
Dual vr controllers gui support

### PR DESCRIPTION
**Pull request by TrevorDev**
PR contains:
- Ability to use multiple vr controllers concurrently on separate gui elements (http://localhost:1338/Playground/index-local.html#ZI9AK7#214)
- Changed behavior so dual controllers are always used (I'm hoping the previous behavior is not desired by existing apps)
- Created new internal VRExperienceHelperGazer class to store gaze information of each controller
- Changes to pointer simulation events to add pointerId
- Removed more pointer singleton logic from control, advancedDynamicTexture classes
- BREAKING CHANGE to onCanvasPointerOutObservable in engine to pass pointerEvent instead of engine in the observable callback. (engine parameter hopefully isnt used in any public apps so I think this is ok?)

Future potential tasks:
- Let multiple curses work on color picker element
- Refactor more webVR helper controller logic into the VRExperienceHelperGazer class

https://github.com/BabylonJS/Babylon.js/issues/3748
https://github.com/BabylonJS/Babylon.js/issues/3653

### CODE REVIEW

File: src/Cameras/VR/babylon.vrExperienceHelper.ts [hunk 1]
```diff
@@ -27,6 +27,145 @@ module BABYLON {
         createFallbackVRDeviceOrientationFreeCamera?: boolean;
     }
 
+    class VRExperienceHelperGazer implements IDisposable {
```
Comment (TrevorDev) [inline src/Cameras/VR/babylon.vrExperienceHelper.ts]: Is this how to correctly create an internal class that should not be accessible publicly?

File: src/Engine/babylon.engine.ts [hunk 2]
```diff
@@ -613,7 +613,7 @@
         /**
          * Observable event triggered each time the canvas receives pointerout event
          */
-        public onCanvasPointerOutObservable = new Observable<Engine>();
+        public onCanvasPointerOutObservable = new Observable<PointerEvent>();
```
Comment (deltakosh) [inline src/Engine/babylon.engine.ts]: I'm fine with the change. Please just add it in the breaking changes list in the what'snew.md

File: src/babylon.scene.ts [hunk 3]
```diff
@@ -1294,9 +1295,11 @@
         /**
          * Use this method to simulate a pointer down on a mesh
          * The pickResult parameter can be obtained from a scene.pick or scene.pickWithRay
+         * @param pickResult pickingInfo of the object wished to simulate pointer event on
+         * @param pointerEventInit pointer event state to be used when simulating the pointer event (eg. pointer id for multitouch)
          */
-        public simulatePointerDown(pickResult: PickingInfo): Scene {
-            let evt = new PointerEvent("pointerdown");
+        public simulatePointerDown(pickResult: PickingInfo, pointerEventInit: PointerEventInit | undefined): Scene {
```
Comment (RaananW) [inline src/babylon.scene.ts]: What is the difference between `pointerEventInit: PointerEventInit | undefined` and pointerEventInit?: PointerEventInit`

**Comment by davrous** [comment]
Code looks good to me, I knew enabling dual interactions would imply a lot of work, this is confirmed :)

What do you mean by: "Changed behavior so dual controllers are always used" ?

File: src/babylon.scene.ts (hunk 3 above)
Comment (TrevorDev) [reply src/babylon.scene.ts]: Looks to work the same (no idea why I wrote it that way) so I switched it to ? to be consistent.

File: src/Engine/babylon.engine.ts (hunk 2 above)
Comment (TrevorDev) [reply src/Engine/babylon.engine.ts]: added

**Comment by TrevorDev** [comment]
@davrous "Changed behavior so dual controllers are always used"

Currently the behavior is that one controller is activated when you enter vr and toggling laser will disable the other hands laser if its currently on. After this change both controllers are activated by default and toggling laser on one hand has no effect on the other hand.

**Comment by RaananW** [comment]
I am fixing the netlify build error right now

**Pull request merged**